test/e2e/network_segmentation.go

```diff
@@ -2195,31 +2204,59 @@ spec:
   topology: Layer3
   layer3:
     role: Primary
-    subnets: ` + generateCIDRforUDN("10.20.100.0/16", "2014:100:200::0/60")
+    subnets: ` + generateCIDRforUDN(cs, "10.20.100.0/16", "2014:100:200::0/60")
 }
 
-func generateCIDRforUDN(v4, v6 string) string {
+func generateCIDRforUDN(cs clientset.Interface, v4, v6 string) string {
 	cidr := `
     - cidr: ` + v4 + `
 `
-	if isIPv6Supported() && isIPv4Supported() {
+	if isIPv6Supported(cs) && isIPv4Supported(cs) {
 		cidr = `
     - cidr: ` + v4 + `
     - cidr: ` + v6 + `
 `
-	} else if isIPv6Supported() {
+	} else if isIPv6Supported(cs) {
 		cidr = `
     - cidr: ` + v6 + `
 `
 	}
 	return cidr
 }
 
-func generateCIDRforClusterUDN(v4, v6 string) string {
+func filterDualStackCIDRs(cs clientset.Interface, cidrs udnv1.DualStackCIDRs) udnv1.DualStackCIDRs {
+	filteredCIDRs := make(udnv1.DualStackCIDRs, 0, len(cidrs))
+	for _, cidr := range cidrs {
+		isIPv6 := utilnet.IsIPv6CIDRString(string(cidr))
+		if isIPv4Supported(cs) && !isIPv6 {
+			filteredCIDRs = append(filteredCIDRs, cidr)
+		}
+		if isIPv6Supported(cs) && isIPv6 {
+			filteredCIDRs = append(filteredCIDRs, cidr)
+		}
+	}
```

Comment:
Let's at least extract this to a helper function we use a l2 and l3

```golang
func isFamilySupported(cs clientset.Interface, cidr string) bool {
    isIPv6 := utilnet.IsIPv6CIDRString(string(cidr))
    return (isIPv4Supported(cs) && !isIPv6)  || (isIPv6Supported(cs) && isIPv6)
}



```
Then we can do something like this at both l2 and l3

```golang
for _, cidr := range cidrs {
                if isFamilySupported(cs, string(cidr)) {
                    filteredCIDRs = append(filteredCIDRs, cidr)
                }
	}
```